### ISSUE TITLE
memory leak SSE after 5000 pub sub event source subscriptions

### ISSUE THREAD

**Issue by delleceste** [slact/nchan#623]
A client subscribing 5000 event sources on nginx + nchan pubsub configuration soon causes nginx being killed because eating all available memory.
The memory consumption grows inexorably over time, not during the subscription phase, rather afterwards when operating, while data is delivered through the channels.

Relevant section of `server' nginx.conf:

server {
	    listen	     18001;
	    listen       18443 ssl;
	    server_name  pwma-dev.elettra.eu;

	    ssl_certificate /etc/ssl/certs/pwma-dev.elettra.eu.bundle.crt;
	    ssl_certificate_key  /etc/ssl/private/pwma-dev_elettra_eu.key;

	    location ~ /sub/(.+)$ {
		nchan_channel_id $1;
		nchan_subscriber;
	      }

	    location ~ /pub/(.+)$ {
	       nchan_publisher;
	       nchan_channel_id $1;
	     }

	  # nginx reverse proxy
	  # bundled requests only are supported
	    location ~ /bu/(.+)$ {
		proxy_set_header Host $host;
		proxy_set_header X-Real-IP $remote_addr;
		proxy_set_header X-Source $1;
	     	#
		# 
		proxy_pass http://caserver-dev;

		# see comments above
		proxy_next_upstream error timeout non_idempotent;

		proxy_connect_timeout 320s;
		proxy_read_timeout 320s;
	    }
	} # server

---------------------------------------------------
NOTES:
---------------------------------------------------
a. worker processes: 1
b. adding the option 

       proxy_buffering off;

in the server section does not help.

=============================================================================================
SYSTEM INFO

$ uname -aro
Linux pwma-dev 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
4GB RAM

* nginx-1.21.1
* nchan-1.2.12

Please find the full nginx.conf attached.

[nginx.zip](https://github.com/slact/nchan/files/7538402/nginx.zip)

**Comment by slact** [comment]
On average, how large are the messages being published? What's the messaging rate per channel?

**Comment by delleceste** [comment]
Hello. Thanks for Your reply.
After quite some time I can get back to the issue.
Is there some integrated tool/option to perform the measurements You required?
Otherwise i shall implement them by myself and let you know.

**Comment by delleceste** [comment]
Hello! Any suggestion?

**Comment by slact** [comment]
[Channel groups](https://nchan.io/#limits-and-accounting) can provide you info on how much memory is being used by the group. Beyond that, you will need to know how much data per channel you are publishing, and how often.

**Comment by delleceste** [comment]
Hello again.
I've just modified the client to provide some statistics about the channel publishing rate.
As long as I ran it on top of latest nchan from git and nginx (1.23.1) I've noticed there is no relevant leak, at least during my first tests (I am not totally sure they are the same (or in the same conditions) as they where at the moment I opened the issue though, need to test over again). 

I see recent activity on the nchan project. 
Is it possible that your latest updates have successfully addressed the issue?

Thanks.
